### DISCUSSION TITLE
Loading indicator

### DISCUSSION THREAD

**Discussion by divdax** [wire-elements/modal#129] [Ideas]
Great Livewire component! I think there should be a loading indicator right after you emit openModal (or with a slight delay for 200-300ms to prevent flickering). When you have a slow server response the user knows there is something going on he has to wait for.

**Comment by martin-ro** [comment]
Have you found a custom solution for this?

**Reply by mityukov** [reply]
I've done something like this:
```html

    <div
        id="loading_overlay"
        class="absolute inset-0 bg-gray-500/75 backdrop-blur-sm z-50 hidden"
    >Loading…</div>

   ...

    <script>
        Livewire.hook('message.sent', (message, omponent) => {
            if (message.updateQueue[0].payload.event === 'openModal') {
                document.getElementById('loading_overlay').classList.add('flex-center');
            }
            if (message.updateQueue[0].payload.method === 'resetState') {
                document.getElementById('loading_overlay').classList.remove('flex-center');
            }
        });
    </script>
```

Though, I wish there was a built-in function. Otherwise, when popup is slow, the users think they need to click it few more times =)

**Reply by mityukov** [reply]
p.s.: `flex-center` is not a part of tailwindcss. It's defined as `@apply flex items-center justify-center`
